### ISSUE TITLE
Typescript build error

### ISSUE THREAD

**Issue by patgaddis** [teijo/jquery-bracket#128]
I am getting an error when trying to compile the typescript, and I was hoping someone could point me in the right direction...

The source is freshly cloned from the github repo, but when I try to compile the jquery.bracket.ts in the src folder, I get the following:

**jquery.bracket.ts(66,9): error TS2417: Class static side 'typeof Score' incorrectly extends base class static side 'typeof Option'.
  Types of property 'of' are incompatible.
    Type '(val: number) => Option<number>' is not assignable to type '<A>(value: A) => Option<A>'.
      Types of parameters 'val' and 'value' are incompatible.
        Type 'A' is not assignable to type 'number'.**

When I had it running under 'npm start', I got the same error, with an additional bit above it about it using tsc v2.5.3.

Any ideas? My apologies if this is something already documented somewhere and I just missed it, but I checked everywhere I could think of. Either way, thanks!

**Comment by teijo** [comment]
New TypeScript compiler must have new stricter or more comprehensive type checking that breaks essentially this:

```typescript
class Super<A> {
  static func<A>(val: A) { }
}
class Sub extends Super<number> {
  static func(val: number) { } // (number) => void not assignable to (A) => void
}
```
It can be circumvented like this:
```typescript
class Fix<N = number> extends Super<N> {
  static func<N>(val: N) { }
}
```
But I'm not sure if that's really how it *should* be done.

So options are: use older TypeScript (`2.0.X`?), do the fix/hack above, or figure out what the real fix is (which might be the above one or not).

**Comment by teijo** [comment]
Fixed in a181e715df1d16bd8d9c138c10ec6bf25b1eb614.
